src/Microsoft.Extensions.DependencyInjection.Specification.Tests/DependencyInjectionSpecificationTests.cs

```diff
@@ -680,5 +680,29 @@ public void DisposesInReverseOrderOfCreation()
             Assert.Equal(outer.MultipleServices.Reverse(), callback.Disposed.Skip(1).Take(3).OfType<IFakeMultipleService>());
             Assert.Equal(outer.SingleService, callback.Disposed[4]);
         }
+
+        [Fact]
+        public void ResolvesMixedOpenClosedGenericsAsEnumerable()
```

Comment:
Yep. That was expected (https://github.com/aspnet/DependencyInjection/issues/500#issuecomment-288051146). I'll see if I can do some magic internally to get it to work.